### PR TITLE
U/jrbogart/write fluxes speedup

### PR DESCRIPTION
**Pull request by JoanneBogart**
The branch u/jrbogart/write_fluxes_speedup is based on u/jrbogart/write_fluxes. Additional features, in order of decreasing importance and complexity, are
1. Support for parallel processing when calculating galaxy fluxes in catalog_creator.py. That part of `_create_galaxy_flux_pixel(..)` which does the calculation is broken out into static routine `_do_galaxy_flux_chunk`. On cori the code runs about 20 times faster using this technique.
2. Add command-line arguments `--flux-only`, `--main-only`, and `--overwrite-data` to `create_sc.py`   to give caller finer control over what will get done.  Add command-line argument `--flux-parallel` to specify # of subprocesses to create in 1.
3. Add entries to config to document magnitude cuts

### CODE REVIEW

**Review by jchiang87 — CHANGES_REQUESTED** [review]
I have handful of comments/suggestions.  The substantive ones arose from trying to run  `create_sc.py --flux-only` with existing skyCatalogs files.

File: python/desc/skycatalogs/catalog_creator.py [hunk 1]
```diff
@@ -392,26 +398,164 @@ def _sed_disk_key(s):
...
+
+        from desc.skycatalogs import open_catalog, SkyCatalog
+
+        self._gal_flux_schema = make_galaxy_flux_schema(self._logname)
+
+        if not config_file:
+            #config_file = self._written_config
+            config_file = self.write_config(path_only=True)
+        if not self._cat:
+            self._cat = open_catalog(config_file)
```
[the remark this answers is not in the document]
Comment (JoanneBogart) [reply python/desc/skycatalogs/catalog_creator.py:437]: I expect so.  I only exercised this code with the environment variable SKYCATALOG_ROOT set which is probably why it worked for me.

File: python/desc/skycatalogs/catalog_creator.py [hunk 2]
```diff
@@ -433,7 +577,8 @@ def create_pointsource_catalog(self, star_truth=None, sn_truth=None):
         self._star_truth = _star_db
         self._sn_truth = _sn_db
 
-        arrow_schema = _make_star_schema()
+        arrow_schema = make_star_schema()
+        self._ps_flux_schema = make_star_flux_schema(self._logname)
```
[the remark this answers is not in the document]
Comment (JoanneBogart) [reply python/desc/skycatalogs/catalog_creator.py:581]: Thanks for the comments.  I'll make at least most, possibly all, of the suggested changes  (there are a couple I have to look at more carefully). I also plan to add a call to initialize of bp path earlier, before creating the subprocesses, so the message doesn't show up multiple times.

File: python/desc/skycatalogs/skyCatalogs.py [hunk 3]
```diff
@@ -261,7 +302,7 @@ def get_objects_by_region(self, region, obj_type_set=None, datetime=None):
         if self.verbose:
             print("Region ", region)
             print("obj_type_set ", obj_type_set)
-        if self._config['area_partition']['type'] == 'healpix':
+        if self._config['area_partition']['partition_type'] == 'healpix':
```
[the remark this answers is not in the document]
Comment (JoanneBogart) [reply python/desc/skycatalogs/skyCatalogs.py:305]: There is no real need for it and I didn't make it consistently. I will change it back.

**Comment by JoanneBogart** [comment]
This is ready for another look, @jchiang87

**Review by jchiang87 — APPROVED** [review]
LGTM

**Pull request merged**